go/libraries/doltcore/sqle/reflog_table_function.go

```diff
@@ -211,14 +225,28 @@ func (rltf *ReflogTableFunction) Expressions() []sql.Expression {
 }
 
 func (rltf *ReflogTableFunction) WithExpressions(expression ...sql.Expression) (sql.Node, error) {
-	if len(expression) > 1 {
-		return nil, sql.ErrInvalidArgumentNumber.New(rltf.Name(), "0 or 1", len(expression))
+	if len(expression) > 2 {
+		return nil, sql.ErrInvalidArgumentNumber.New(rltf.Name(), "0 to 2", len(expression))
 	}
 
 	new := *rltf
-	if len(expression) > 0 {
-		new.refExpr = expression[0]
+
+	if len(expression) == 2 {
+		if expression[0].String() == "'--all'" && expression[1].String() == "'--all'" {
+			return nil, fmt.Errorf("error: multiple values provided for `all`")
+		}
+		if expression[0].String() != "'--all'" && expression[1].String() != "'--all'" {
+			return nil, fmt.Errorf("error: %s has too many positional arguments. Expected at most %d, found %d: %s", rltf.Name(), 1, 2, expression)
+		}
 	}
```

Comment:
Calling `expression[0].String()` and looking at the literal string value of the expression is going to be buggy. It works for basic cases, but it's fragile and will break. For example, it's totally valid to pass in `"--all"` in addition to `'--all'`. It's also possible that customers may want to pass in a SQL variable as the expression, and have that SQL variable hold the argument. There's lots of these potential edge cases, and it's better to make this code resilient from the start, instead of having to come back to it when a customer hits one of these. 

The best way to do this is to have the `WithExpressions` function simple assign the expressions as is to the new rftf instance and don't try to evaluate or directly inspect the expression values here. Do that evaluation and argument validation as part of the `PartitionRows` code – that's when we know the query plan will be fully resolved and it's safe to evaluate the expressions. (Technically... it's safe to evaluate the expressions whenever `Resolved()` returns true, but if you do that here, you'll still have to do it in the `PartitionsRows` function in case something wasn't resolved yet here, so it doesn't really make sense to do that anywhere else except for inside `PartitionsRows`).